### DISCUSSION TITLE
How to inherit default  connection parameters defined in swanctl.conf in my program

### DISCUSSION THREAD

**Discussion by vishal1kumar** [strongswan/strongswan#416] [General]
I am writing a python code to configure connection details in strongswan using vici interface. I have some default connection parameters defined in swanctl.conf file as described in link: https://wiki.strongswan.org/projects/strongswan/wiki/strongswanconf
```
swanctl.conf contains:
child-defaults {
     # defaults for child configs (e.g. traffic selectors)
     rekey_time = 180
}

```
Now in my python code while configuring new locations I want to inherit these default parameters for all new connection:
 
```
loc_1 =  {
            'remote_addrs': [],
            'mobike': 'no',
            'local' : {
                'id': 'ipsec.westus.cloudapp.azure.com',
                'auth': 'psk'
            },
            'children' : {
                'child-sa' : {
                    'remote_ts': [],
                    'mark_in': 0,
                    'mark_out': 0,
                    'local_ts': ['0.0.0.0/0[tcp/http]', '0.0.0.0/0[tcp/https]'],
                }
            },
            'remote' : {
                'id': '',
                'auth': 'psk'
            }
        }

```
**Question: How should I inherit 'child-defaults' for 'child-sa' for my loc_1?**

**Comment by Thermi** [comment]
Hi, you need to resolve all preprocessor directives like that includes oder references to objects in your software. VICI only takes complete objects. It doesn't do any further processing.

Am May 26, 2021 5:42:17 PM UTC schrieb vishal1kumar ***@***.***>:
>I am writing a python code to configure connection details in
>strongswan using vici interface. I have some default connection
>parameters defined in swanctl.conf file as described in link:
>https://wiki.strongswan.org/projects/strongswan/wiki/strongswanconf
>```
>swanctl.conf contains:
>child-defaults {
>     # defaults for child configs (e.g. traffic selectors)
>     rekey_time = 180
>}
>
>```
>Now in my python code while configuring new locations I want to inherit
>these default parameters for all new connection:
> 
>```
>loc_1 =  {
>            'remote_addrs': [],
>            'mobike': 'no',
>            'local' : {
>                'id': 'ipsec.westus.cloudapp.azure.com',
>                'auth': 'psk'
>            },
>            'children' : {
>                'child-sa' : {
>                    'remote_ts': [],
>                    'mark_in': 0,
>                    'mark_out': 0,
>           'local_ts': ['0.0.0.0/0[tcp/http]', '0.0.0.0/0[tcp/https]'],
>                }
>            },
>            'remote' : {
>                'id': '',
>                'auth': 'psk'
>            }
>        }
>
>```
>**Question: How should I inherit 'child-defaults' for 'child-sa' for my
>loc_1?**
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/416

-- 
Sent from mobile

**Reply by vishal1kumar** [reply]
Thanks for the response

**Reply by Thermi** [reply]
Is that issue solved?

**Reply by vishal1kumar** [reply]
Yes, I am now passing the default values for al connections in vici: "load-conn".

I dont see the option in this query to mark it 'answered'..
